alerts/AggregatedAPIErrors.md

```diff
@@ -0,0 +1,65 @@
+v1.build.openshift.io                           openshift-apiserver/api                                      True        340d
+v1.image.openshift.io                           openshift-apiserver/api                                      True        340d
+v1.oauth.openshift.io                           openshift-apiserver/api                                      True        340d
+v1.packages.operators.coreos.com                openshift-operator-lifecycle-manager/packageserver-service   True        340d
+v1.project.openshift.io                         openshift-apiserver/api                                      True        340d
+v1.quota.openshift.io                           openshift-apiserver/api                                      True        340d
+v1.route.openshift.io                           openshift-apiserver/api                                      True        340d
+v1.security.openshift.io                        openshift-apiserver/api                                      True        340d
+v1.template.openshift.io                        openshift-apiserver/api                                      True        340d
+v1.user.openshift.io                            openshift-apiserver/api                                      True        340d
+v1beta1.metrics.k8s.io                          openshift-monitoring/prometheus-adapter                      True        340d
+```
+Here we want the availability status to be `True`. A `False` means that requests for that API service, API server pods, or resources belonging to that apiGroup failed many times during the last minutes.
+
+##### Check the `apiserver` pods
+```shell
+$ oc get pods -n openshift-apiserver
+NAME                        READY   STATUS    RESTARTS   AGE
+apiserver-c74d6bd79-pkmw4   1/1     Running   0          41d
+apiserver-c74d6bd79-t4l9d   1/1     Running   0          41d
+apiserver-c74d6bd79-z6qxx   1/1     Running   0          41d
+```
+If their status is not `Running` check the logs for more details. 
+
+##### Check the authentication certificates of the aggregated API
+Make sure the certificates are up to date and still valid. Use:
+```shell
+$ oc get configmaps -n kube-system extension-apiserver-authentication
+```
+you can save those certificates into a file and use the following command to check the end dates:
+```shell
+$ openssl x509 -noout -enddate -in {myfile_with_certs.crt}
+```
+Those certificates are used by the aggregated APIs to validate requests.
+
+## Mitigation
+In case a certificate expiry can be excluded but you observed some hanging pods, try to find issues in the logs. However, as these pods are controlled by a deployment, you can restart them when they are not answering requests.
+
+**TODO more explanations/commands**
+
+
+## References
+* https://docs.openshift.com/container-platform/latest/security/certificates/api-server.html
+* https://github.com/openshift/cluster-monitoring-operator
```

Comment:
I'm not super-excited about a references section at the end of the doc.  Why wouldn't we link these from the relevant sections?  E.g. folks checking auth certs might want a link from that section to the official docs around that.  Folks who are looking at APIServices probably don't care about official auth cert docs at that point.

And it's not clear to me what part of this runbook would lead me to productively click through to the monitoring repository.  If we have particular content in that repository that this runbook draws from (e.g. the alert definition?), we should link directly too it, like:

    [This alert][AggregatedAPIErrors] is triggered...
    ...
    [AggregatedAPIErrors]: https://github.com/openshift/cluster-monitoring-operator/blob/aefc8fc5fc61c943dc1ca24b8c151940ae5f8f1c/assets/control-plane/prometheus-rule.yaml#L440-L449

That pins a particular version of the operator, which would be great if we were baking these runbooks into the release image.  But even if we keep them separate, it still gives folks some guidance, and folks who are motivated enough can use that to track down exactly the version that corresponds to their cluster.

Reply:
It totally makes sense to directly link from the appropriate sections. I will include that. And the link to the Prometheus rule would always target the latest version in master. But it's unclear for me how we warn the reader that he may have a different version/setting deployed for that alert. Let's trust here the common sense instead of mentioning it in every single alert runbook and see how it goes.